internal/worker/apiremotecaller/worker.go

```diff
@@ -218,8 +216,16 @@ func (w *remoteWorker) loop() error {
 
 			w.cfg.Logger.Debugf(ctx, "remote workers updated: %v", required)
 
+			var remotes []RemoteConnection
+			for _, server := range required {
+				if server == nil {
+					continue
+				}
+				remotes = append(remotes, server)
+			}
+
 			w.mu.Lock()
-			w.apiRemotes = slices.Collect(maps.Values(required))
+			w.apiRemotes = remotes
```

Comment:
Sorry, why did this change? Can `newRemoteServer` return a nil value without an error??

Reply:
To fix a racy test. This is going to change real soon, to not use pubsub. Whilst that change is happening, I don't want this to fail for people.